hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -85,7 +86,9 @@ public void validate(YamlMapping rootNode) {
             // this could be expressed in the schema as well, but that would make all the schema validation errors much harder
             // to read, so it is better to implement it here as a semantic check
             List<String> definedRootNodes = PERMITTED_ROOT_NODES.stream()
-                    .filter(rootNodeName -> rootNode != null && rootNode.child(rootNodeName) != null)
+                    .filter(rootNodeName -> rootNode != null
+                            && StreamSupport.stream(rootNode.childrenPairs().spliterator(), false)
+                                            .anyMatch(pair -> pair.nodeName().equals(rootNodeName)))
```

Comment:
Will this be correct with 
```
hazelcast:
  hazelcast: ...
```
?

Reply:
Former version was looking for a root element with a value (see that it is checking for the value of node in `rootNode.child(rootNodeName)`), I changed it to only check for the existence of keys.

Btw, given example will fail on the schema validation phase, it's not valid.